tests/subsys/canbus/isotp/conformance/src/main.c

```diff
@@ -1106,7 +1124,7 @@ void *isotp_conformance_setup(void)
 	zassert_true(device_is_ready(can_dev), "CAN device not ready");
 
 	ret = can_set_mode(can_dev, CAN_MODE_LOOPBACK |
-			   (IS_ENABLED(CONFIG_ISOTP_USE_CAN_FD) ? CAN_MODE_FD : 0));
+			   (IS_ENABLED(CONFIG_CAN_FD_MODE) ? CAN_MODE_FD : 0));
```

Comment:
In that case `IS_ENABLED(CONFIG_CAN_FD_MODE)` should evaluate to `0` and FD would not be enabled.

Or do you have a case in mind where we'd have one controller with FD capabilities and one without on the same board/SOC, so that we have `CONFIG_CAN_FD_MODE=y` even though one controller doesn't support it?

Reply:
`CONFIG_CAN_FD_MODE` can be enabled whether the CAN controller supports CAN-FD or not.

Reply:
Yeah, that's exactly how the test is run at the moment, e.g. with `native_posix`. The first case is with default setup `CONFIG_CAN_FD_MODE=n` (even though it would be supported) and the two other cases with extra config `CONFIG_CAN_FD_MODE=y` and some different max data length settings.

Or maybe I misunderstand your question/request?

Reply:
The tests need to be able to run on any CAN controller, regardless of `CONFIG_CAN_FD_MODE` being `y` or `n` and whether the CAN controller driver supports CAN-FD or not.

I'd suggest using `can_get_capabilities()` to determine if `CAN_MODE_FD` should be passed to `can_set_mode()` . That does not require any `#ifdef` on `CONFIG_CAN_FD_MODE`.

Reply:
It was slightly more complicated than just checking with `can_get_capabilities` and applying the mode accordingly. Now we have the existing `main` test suite, which is skipped if FD is enabled for a controller that does not have FD capabilities. To check that correct errors are returned in such cases, I added another test suite `canfd_notsup.c`.